include/fi_rbuf.h

```diff
@@ -246,7 +246,7 @@ static inline int ofi_rbfdinit(struct ofi_ringbuffd *rbfd, size_t size)
 	ofi_close_socket(rbfd->fd[1]);
 err1:
 	ofi_rbfree(&rbfd->rb);
-	return -errno;
+	return -ofi_sockerr();
```

Comment:
We can reach here if fi_fd_nonblock() fails.  Since fi_fd_nonblock() is already an abstracted call, it should return -errno on failure.  Let's limit ofi_sockerr() use to only the case where socketpair() call fails. E.g.
```
ret = socketpair(...);
if (ret == SOCKET_ERROR) {
    ret = -ofi_sockerr();
   goto err1;
}

ret = fi_fd_nonblock(...)
if (ret)
    goto err2;
...
err1:
    return ret;
```

Reply:
👍 Thanks